app/moderation/ModerationPageContent.tsx

```diff
@@ -0,0 +1,1437 @@
+    globallyBannedUsersCount,
+    globallyBannedUsersOffset,
+    showExpiredBans,
+    limit,
+  } = props;
+
+  const classes = useStyles(styles);
+  const searchParams = useSearchParams();
+  const [expandedRows, setExpandedRows] = useState<Set<string>>(new Set());
+
+  // Fetch full comment data via GraphQL
+  const { data: commentsData, loading: commentsLoading } = useQuery(ModeratorCommentsMultiQuery, {
+    variables: { commentIds: moderatorCommentIds },
+    fetchPolicy: 'cache-first',
+    skip: moderatorCommentIds.length === 0,
+  });
+
+  const moderatorComments = commentsData?.comments?.results || [];
+
+  const toggleRowExpanded = (id: string) => {
+    setExpandedRows((prev) => {
+      const next = new Set(prev);
+      if (next.has(id)) next.delete(id);
+      else next.add(id);
+      return next;
+    });
+  };
+
+
+  const buildToggleExpiredRateLimitsUrl = () => {
+    const params = new URLSearchParams(searchParams?.toString());
+    if (showExpiredRateLimits) {
+      params.delete('showExpiredRateLimits');
+    } else {
+      params.set('showExpiredRateLimits', 'true');
+    }
+    // Reset pagination when toggling
+    params.delete('activeRateLimitsOffset');
+    return `/moderation?${params.toString()}`;
+  };
+
+  const buildToggleNewUserRateLimitsUrl = () => {
+    const params = new URLSearchParams(searchParams?.toString());
+    if (showNewUserRateLimits) {
+      params.delete('showNewUserRateLimits');
+    } else {
+      params.set('showNewUserRateLimits', 'true');
+    }
+    // Reset pagination when toggling
+    params.delete('activeRateLimitsOffset');
+    return `/moderation?${params.toString()}`;
+  };
+
+  const buildToggleExpiredBansUrl = () => {
+    const params = new URLSearchParams(searchParams?.toString());
+    if (showExpiredBans) {
+      params.delete('showExpiredBans');
+    } else {
+      params.set('showExpiredBans', 'true');
+    }
+    // Reset pagination when toggling
+    params.delete('globallyBannedUsersOffset');
+    return `/moderation?${params.toString()}`;
+  };
+
+  const getRateLimitCategoryTag = (category: "static" | "rolling" | "timed") => {
+    const tagClass = category === "rolling" ? classes.rollingTag :
+                     category === "timed" ? classes.timedTag :
+                     classes.staticTag;
+
+    const tooltips = {
+      rolling: "Based on last 20 posts/comments - never expires, updates as contributions change",
+      timed: "Based on 30-day window - can expire as time passes",
+      static: "Based on total karma - always active while karma condition is met"
+    };
+
+    return (
+      <span
+        className={`${classes.rateLimitCategoryTag} ${tagClass}`}
+        title={tooltips[category]}
+      >
+        {category}
+      </span>
+    );
+  };
+
+  const renderContent = (contents: any) => {
+    if (!contents?.html) return null;
+    return <div className={classes.expandedContent} dangerouslySetInnerHTML={{ __html: contents.html }} />;
```

Comment:
don't we have a dedicated component for this?

Reply:
Yeah, but I don't actually want the styling of that

Reply:
k